feat(perf): Requests module SDK instrumentation documentation (#9963)
getsentry/sentry-docs: docs/product/performance/requests.mdx

  ></iframe>
</div>

## Prerequisites and Limitations

<Note>
Network requests using non-HTTP protocols (FTP, WebSocket, etc.) are not supported at this time.
</Note>

Availability of HTTP request monitoring depends on the SDK your application uses. In most cases, Sentry's SDKs automatically enable HTTP request tracking. You can check your SDK's <PlatformLink to="/performance/instrumentation/automatic-instrumentation">automatic instrumentation documentation</PlatformLink> to see if it includes tracking HTTP requests.

> 

Reviewer: NIt, I might add a heading "Custom Instrumentation" to draw people's attention to this option.

Also, worth a discussion about whether we should suggest setting `network.protoco.name`, `network.protocol.version`, if people try to instrument an HTTP/2 connection?

Author: 🤔 I can add this to our span data conventions!

> If the SDK you are using doesn't automatically instrument HTTP requests, you can use <PlatformLink to="/performance/instrumentation/requests-module/">custom instrumentation</PlatformLink> instructions to instrument your application. If your platform doesn't have dedicated custom instrumentation instructions, try adapting the code from another platform, since all the concepts are the same.

Reviewer 2: ```suggestion
If the SDK you're using doesn't automatically instrument HTTP requests, you can instrument your application by following the <PlatformLink to="/performance/instrumentation/requests-module/">custom instrumentation</PlatformLink> instructions. Because all the concepts are the same, if your platform doesn't have dedicated custom instrumentation instructions, you can try adapting the code from another platform.
```
